math/mathcore/inc/Math/GenAlgoOptions.h

```diff
@@ -84,6 +85,27 @@ class GenAlgoOptions : public IOptions {
       InsertValue(name, fNamOpts, std::string(val));
    }
 
+   std::vector<std::string>  GetAllNamedKeys() {
+      std::vector<std::string> names;
+      // start by named options
+      for (auto & e : fNamOpts) 
+         names.push_back(e.first.c_str());
```

Comment:
```C++
      std::vector<std::string> names;
      names.reserve(fNamOpts);
      // start by named options
      for (auto const& e : fNamOpts) 
         names.emplace(e.first);
```
* `std::string` can be directly copy constructed from other `std::string` without calling `c_str()`
* `auto const&` is preferred over `auto &`
* we know how many elements the output vector we have, so `reserve` can be used

Also the same the two other functions below.

Reply:
Thanks for the suggestion